out_writeapi_test.go

```diff
@@ -38,3 +56,136 @@ func TestFLBPluginRegister(t *testing.T) {
 	assert.Equal(t, "writeapi", currplugin.name)
 
 }
+
+// this is a struct keeping track of whether the correct options are sent in NewManagedStream
+type OptionChecks struct {
+	configProjectID        bool
+	configDatasetID        bool
+	configTableID          bool
+	configMaxChunkSize     bool
+	configMaxQueueSize     bool
+	configMaxQueueRequests bool
+	calledGetClient        bool
```

Comment:
Better to have a count of how many times the function got called.

Reply:
We can do this - but this doesn't check that each function was called (just that it was called the right number of times) - do you think that's sufficient?

Reply:
Ah, but your test will then verify that the function was indeed called the expected number of times. We often use tests like this.